File: CSharp/5kyu/MaximumSubarraySum.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace CSharp._5kyu
{
    public static partial class Kata5
    {
        public static int MaxSequence(int[] arr)
        {
            int max = 0, res = 0, sum = 0;

            foreach (int item in arr)
            {
                sum += item;
                max = sum > max ? max : sum;
                res = res > sum - max ? res : sum - max;
            }

            return res;
        }
        public static int MaxSequence_kata_2(int[] arr)
        {
            int max = 0;
            for (int i = 0; i < arr.Length; i++)
            {
                int sum = 0;
                for (int j = i; j < arr.Length; j++)
                {
                    sum += arr[j];
                    max = Math.Max(sum, max);
                }
            }

            return max;
        }
    }
}

File: Python/5kyu/MaximumSubarraySum.py
def max_sequence(arr):
    c_sum = 0
    max_sum = 0

    for v in arr:
        c_sum = max(c_sum + v, 0)
        max_sum = max(c_sum, max_sum)

    return max_sum


print(max_sequence([-2, 1, -3, 4, -1, 2, 1, -5, 4]))
